domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -722,7 +722,7 @@ class ProfileManagementController @Inject constructor(
   }
 
   private fun onlyLetters(name: String): Boolean {
-    return name.matches(Regex("^[ A-Za-z]+\$"))
+    return name.matches(Regex("^(?!.*^[-.'])^(?!.*[.'-]\$).*^(?!.*[0-9!@#\$%^&*()_+={}\\[\\]|\\\\;\",?/<>`~]).*[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'-]"))
```

Comment:
This pattern has grown to an unmanageable size, I think. :) Can you instead split this into multiple, more specific patterns and check them one-at-a-time? I suggest moving them to fields to provide semantic meaning (such as having one for verifying that disallowed symbols are absent)l.